src/irmin-pack/unix/control_file.ml

```diff
@@ -69,9 +69,8 @@ module Make (Io : Io.S) = struct
 
   let read io =
     let open Result_syntax in
-    let* len = Io.read_size io in
-    let len = Int63.to_int len in
-    let* string = Io.read_to_string io ~off:Int63.zero ~len in
```

Comment:
If the RW instance updates the file between the `read_size` and the `read_to_string` we would have a problem